crates/edr_evm/src/block/builder.rs

```diff
@@ -247,13 +248,21 @@ impl BlockBuilder {
             };
         }
 
+        let spec_id = self.cfg.handler_cfg.spec_id;
         let blob_gas_used = transaction.total_blob_gas().unwrap_or_default();
         if let Some(BlobGas {
             gas_used: block_blob_gas_used,
             ..
         }) = self.header.blob_gas.as_ref()
         {
-            if block_blob_gas_used + blob_gas_used > MAX_BLOB_GAS_PER_BLOCK {
+            let max_blobs_per_block = if spec_id >= SpecId::PRAGUE {
+                eip7691::MAX_BLOBS_PER_BLOCK_ELECTRA
+            } else {
+                eip4844::MAX_BLOBS_PER_BLOCK as u64
```

Comment:
Nit: would read nicer with a comment "usize is guaranteed to fit into u64"

Reply:
Addressed in https://github.com/NomicFoundation/edr/pull/791/commits/86ccde1007aff43060f0f24e3112d61e5b8a7f5a